### PR TITLE
Add documentation on how to use random number generation

### PR DESCRIPTION
**Pull request by fdplacido**
BEGINRELEASENOTES
- Add documentation on how to use random number generation in k4MarlinWrapper and other key4hep components

ENDRELEASENOTES

Needs k4FWCore PR Merged: https://github.com/key4hep/k4FWCore/pull/80

### CODE REVIEW

File: doc/randomNumberGeneration.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+Random number generation can be set from the Gaudi options file, including setting an array of seeds:
+
+```python
+from Configurables import HepRndm__Engine_CLHEP__RanluxEngine_ as RndmEngine
+rndmengine = RndmEngine('RndmGenSvc.Engine')
+rndmengine.SetSingleton = True
+rndmengine.Seeds = [ 2345678 ] # default seed is 1234567
+```
+
+If running a `MarlinProcessorWrapper` the seed set for the engine will be directly set for the underlying Marlin processor. As in the previous source code snippet, if the seed used is `2345678`, Marlin will receive that same seed.
```
Comment (andresailer) [inline doc/randomNumberGeneration.md:12]:
```suggestion
If running a `MarlinProcessorWrapper` the seed set for the engine will be directly set for the underlying Marlin EventSeeder object. As in the previous source code snippet, if the seed used is `2345678`, Marlin will receive that same seed.
```

File: doc/randomNumberGeneration.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+```cpp
+StatusCode SomeGaudiAlgorithm::initialize() {
+  m_service = serviceLocator()->service("UniqueIDGenSvc");
+```
+
+Then use the service during execution:
+
+```cpp
+StatusCode MarlinProcessorWrapper::execute() {
+  m_thing->getUniqueID(1,2,"SomeClusterAlgorithm");
```
Comment (andresailer) [inline doc/randomNumberGeneration.md:63]: what is `m_thing`?

Comment (andresailer) [reply doc/randomNumberGeneration.md:63]:
```suggestion
  m_thing->getUniqueID(1, 2, name());
```
I think this is the only interface we should support?

**Comment by andresailer** [comment]
Rebase and modifications for https://github.com/key4hep/k4FWCore/pull/83 needed

File: doc/randomNumberGeneration.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
+# Random number generation
+
+Random number generation can be set from the Gaudi options file, including setting an array of seeds:
```
Comment (andresailer) [inline doc/randomNumberGeneration.md:3]: This first part should really be dropped now, and only the UniqueIDGenSvc should be advertised.
And probably this should go to k4FWCore

Comment (fdplacido) [reply doc/randomNumberGeneration.md:3]: Yeah I agree, I will close this PR and open one there with only the UniqueIDGenSvc part.

**Pull request closed without merging**